packages/unified-latex-to-pretext/tests/unified-latex-macro.test.ts

```diff
@@ -0,0 +1,76 @@
+import { describe, it, expect } from "vitest";
+import Prettier from "prettier";
+import util from "util";
+import { getParser } from "@unified-latex/unified-latex-util-parse";
+import { report_macros, expand_user_macros } from "../libs/report-and-expand-macros";
+import { printRaw } from "@unified-latex/unified-latex-util-print-raw";
+
+// Make console.log pretty-print by default
+const origLog = console.log;
+console.log = (...args) => {
+    origLog(...args.map((x) => util.inspect(x, false, 10, true)));
+};
+
+describe("unified-latex-to-pretext:report-and-expand-macro", () => {
+    let value: string;
+
+    it("can reported unsupported macros", () => {
+        value = String.raw`$\mathbb{R} \fakemacro{X}$`;
+
+        const parser = getParser();
+        const ast = parser.parse(value);
+
+        expect(report_macros(ast)).toEqual(["fakemacro"]);
+    });
+
+    it("can report no unsupported macros in mathmode", () => {
+        value = String.raw`$\mathbb{R} \frac{1}{2} \cup$`;
+
+        const parser = getParser();
+        const ast = parser.parse(value);
+
+        expect(report_macros(ast)).toEqual([]);
+    });
+
+    it("can report no unsupported macros not in mathmode", () => {
+        value = String.raw`\underline{text} \textbf{bold} \subsection{section}`;
+
+        const parser = getParser();
+        const ast = parser.parse(value);
+
+        expect(report_macros(ast)).toEqual([]);
+    });
```

Comment:
Need more tests. 
1. Unsupported macros outside of math mode.
2. Unsupported macros in text mode with a math anscestor should be reported. E.g. `$\frac{1}{\text{ hi \foo}}$`